libs/blocks/mas-autoblock/mas-autoblock.js

```diff
@@ -22,32 +50,162 @@ export function getTagName(el) {
+  if (!hierarchy) return null;
+
+  const sidenav = createTag('merch-sidenav', { sidenavTitle: 'Categories' });
+
+  /* Search */
+  const searchText = control.data?.placeholders?.searchText || 'Search all products';
+  if (searchText) {
+    const spectrumSearch = createTag('sp-search', { placeholder: searchText });
+    const search = createTag('merch-search', { deeplink: 'search' });
+    search.append(spectrumSearch);
+    sidenav.append(search);
+  }
+
+  /* Filters */
+  const spSidenav = createTag('sp-sidenav', { manageTabIndex: true });
+  spSidenav.setAttribute('manageTabIndex', true);
+  const sidenavList = createTag('merch-sidenav-list', { deeplink: 'filter' }, spSidenav);
+
+  let multilevel = false;
+  function generateLevelItems(level, parent) {
+    for (const node of level) {
+      const value = node.label.toLowerCase();
+      const item = createTag('sp-sidenav-item', { label: node.label, value });
+      parent.append(item);
+      if (node.collections) {
+        multilevel = true;
+        generateLevelItems(node.collections, item);
+      }
+    }
+  }
+
+  generateLevelItems(hierarchy, spSidenav);
+  if (multilevel) spSidenav.setAttribute('variant', 'multilevel');
+
+  sidenav.append(sidenavList);
+
+  return sidenav;
+}
+
+export async function createControl(el, options, tagName) {
+  const [attributes, html] = getTagAttributesAndContent(options.fragment, tagName);
+  const control = createTag(tagName, attributes, html);
+  let element = control;
+
+  switch (tagName) {
+    case 'merch-card-collection':
+      if (options.sidenav) {
+        const container = createTag('div', null, control);
+        element = container;
+      }
+      break;
+    default:
+      break;
+  }
+
+  el.replaceWith(element);
+  await checkReady(control);
+
+  switch (tagName) {
+    case 'merch-card-collection': {
+      /* Placeholders */
+      const placeholders = control.data?.placeholders || DEFAULT_PLACEHOLDERS;
+      for (const key of Object.keys(placeholders)) {
+        const value = placeholders[key];
+        const tag = value.includes('<p>') ? 'div' : 'p';
+        const placeholder = createTag(tag, { slot: key }, value);
+        control.append(placeholder);
+      }
+
+      /* Sidenav */
+      if (options.sidenav) {
+        const sidenav = getCollectionSidenav(control);
+        if (!sidenav) break;
+        element.insertBefore(sidenav, control);
+        control.sidenav = sidenav;
+        control.requestUpdate();
+      }
+
+      element.classList.add(`${control.variant}-container`);
+
+      break;
+    }
+    default:
+      break;
   }
 }
 
 export default async function init(el) {
-  const fragment = getFragmentId(el);
-  if (!fragment) return;
-  await createCard(el, fragment);
+  const options = getOptions(el);
+  if (!options.fragment) return;
+  const tagName = getTagName(el);
+  await loadControl(tagName);
+  await createControl(el.closest('.content'), options, tagName);
```

Comment:
```suggestion
await loadDependencies(tagName);
await createMasElements(el.closest('.content'), options, tagName);
```

Reply:
when i read 'control' i get no idea of what is it, mas-commerce-service or merch-card-collection or swc libs..